Tasgrid/tasgridWrapper.cpp

```diff
@@ -600,6 +626,37 @@ bool TasgridWrapper::getIntegrate(){
 
     return true;
 }
+bool TasgridWrapper::getDifferentiate(){
+    if (grid.getNumLoaded() == 0){
+        cerr << "ERROR: no values loaded in the grid, cannot evaluate!" << endl;
+        return false;
+    }
+    if (grid.getNumOutputs() == 0){
+        cerr << "ERROR: no outputs set for the grid, nothing to evaluate!" << endl;
+        return false;
+    }
+    auto x = readMatrix(xfilename);
+    if (x.getStride() != (size_t) grid.getNumDimensions()){
+        cerr << "ERROR: grid is set for " << grid.getNumDimensions() << " dimensions, but " << xfilename << " specifies " << x.getStride() << endl;
+        return false;
+    }
+    if (x.empty()){
+        cerr << "ERROR: no points specified in " << xfilename << endl;
+        return false;
+    }
+    int num_points = x.getNumStrips();
+    int num_in = grid.getNumDimensions();
+    int num_out = grid.getNumOutputs();
+    Data2D<double> result(num_in * num_out, num_points);
+    #pragma omp parallel for
+    for (int i=0; i<num_points; i++)
+        grid.differentiate(x.getStrip(i), result.getStrip(i));
+
+    writeMatrix(outfilename, num_points, num_out * num_in, result.data());
```

Comment:
The indexing is much better, but it is still prone to error (even in the rest of the code).

A better way to do it is to add overloads to `writeMatrix()` and `printMatrix()` that accept `Data2D<T>` and no sizes, then the overload will call the main method by taking the sizes from the number of strips and the size of the stride.

Reply:
If you do this, you can also skip the declaration of the `int` variables, e.g.,
```
Data2D<double> result(grid.getNumDimensions() * grid.getNumOutputs(), x.getNumStrips());
for (int i=0; i<x.getNumStrips(); i++)  grid.differentiate(x.getStrip(i), result.getStrip(i));
writeMatrix(outfilename, result);
```

Reply:
Changed. I added the overloads and changed  `TasgridWrapper::getDifferentiate()` and `TasgridWrapper::getDiffWeights()` to use them.